app/scripts/modules/core/src/delivery/filter/ExecutionFilters.tsx

```diff
@@ -0,0 +1,287 @@
+import * as React from 'react';
+import * as ReactGA from 'react-ga';
+import { get, orderBy, uniq } from 'lodash';
+import { BindAll, Debounce } from 'lodash-decorators';
+import { $q } from 'ngimport';
+import { SortableContainer, SortableElement, SortableHandle, arrayMove, SortEnd } from 'react-sortable-hoc';
+import { Subscription } from 'rxjs';
+
+import { Application } from 'core/application';
+import { FilterSection } from 'core/cluster/filter/FilterSection';
+import { IPipeline } from 'core/domain';
+import { ReactInjector } from 'core/reactShims';
+
+import './executionFilters.less';
+
+export interface IExecutionFiltersProps {
+  application: Application;
+}
+
+export interface IExecutionFiltersState {
+  pipelineNames: string[];
+  pipelineReorderEnabled: boolean;
+  tags: any[];
+}
+
+const DragHandle = SortableHandle(() => (
+  <span className="pipeline-drag-handle clickable glyphicon glyphicon-resize-vertical"/>
+));
+
+@BindAll()
+export class ExecutionFilters extends React.Component<IExecutionFiltersProps, IExecutionFiltersState> {
+  private executionsRefreshUnsubscribe: () => void;
+  private groupsUpdatedSubscription: Subscription;
+  private locationChangeUnsubscribe: () => void;
+  private pipelineConfigsRefreshUnsubscribe: () => void;
+
+  constructor(props: IExecutionFiltersProps) {
+    const { executionFilterModel } = ReactInjector;
+    super(props);
+
+    this.state = {
+      pipelineNames: this.getPipelineNames(),
+      pipelineReorderEnabled: false,
+      tags: executionFilterModel.asFilterModel.tags,
+    }
+  }
+
+  public componentDidMount(): void {
+    const { application } = this.props;
+    const { $rootScope, executionFilterModel, executionFilterService } = ReactInjector;
+
+    this.executionsRefreshUnsubscribe = application.executions.onRefresh(null, () => { this.refreshPipelines(); });
+    this.groupsUpdatedSubscription = executionFilterService.groupsUpdatedStream.subscribe(() => this.setState({ tags: executionFilterModel.asFilterModel.tags }));
+    this.pipelineConfigsRefreshUnsubscribe = application.pipelineConfigs.onRefresh(null, () => { this.refreshPipelines(); });
+
+    this.initialize();
+
+    this.locationChangeUnsubscribe = $rootScope.$on('$locationChangeSuccess', () => {
```

Comment:
would check with @christopherthielen if there's a better way to handle this in react world